src/MovieDetails/MovieDetails.js

```diff
@@ -0,0 +1,204 @@
+  }
+
+  generateListItem() {
+    return (
+      <section>
+        <ListItem 
+          key={Math.random()}
+          label="Release Date"
+          body={new Date(this.state.singleMovie.release_date).toLocaleDateString()}
+        />
+        <ListItem 
+          key={Math.random()}
+          label="Runtime"
+          body={`${this.state.singleMovie.runtime} mins`}
+        />
+        <ListItem 
+          key={Math.random()}
+          label="Budget"
+          body={`$${new Intl.NumberFormat("en-US").format(this.state.singleMovie.budget)}`}
+        />
+        <ListItem 
+          key={Math.random()}
+          label="Revenue"
+          body={`$${new Intl.NumberFormat("en-US").format(this.state.singleMovie.revenue)}`}
+        />
+      </section>
+    )
+  }
+
+  toggleWatchlist = () => {
+    if (this.state.onWatchlist === false) {
+      this.setState({ onWatchlist: true })
+    } else {
+      this.setState({ onWatchlist: false})
+    }
+  }
+
+
+  render() {
+    return(
+      <section className="movie-details">
+        <section className="banner-container">
+          <img src={this.state.singleMovie.backdrop_path} className="banner-img"/>
+        </section>
+        <section className="movie-info">
+          <section className="movie-aside">
+            <img src={this.state.singleMovie.poster_path} className="poster-img" />
+            {this.generateListItem()}
+          </section>
+          <section className="movie-main">
+            <section className="backdrop-overlay"> 
+              <section className="responsive-poster">
+                <img src={this.state.singleMovie.poster_path} className="poster-img" />
+              </section>
+              <section className="main-header-details">
+                <h1 className="title">{this.state.singleMovie.title}</h1>
+                <p className="tagline">{this.state.singleMovie.tagline}</p>
+                <p className="rating">
+                  <span className="rating-star">☆ {this.state.formattedRating}</span>/10
+                </p>
+                <section className="genre-list">
+                  {this.formatGenres()}
+                </section>
+              </section>  
+            </section>
+            <section className="overview-box">
+              <section className="rating-watchlist">
+                <StarRating 
+                  currentUser={this.props.currentUser}
+                  currentUserRating={!this.state.currentUserRating ? 0 : this.state.currentUserRating.rating}
+                  canEdit={!this.props.currentUser ? false : true} 
+                  setStarRating={this.setStarRating}
+                />
+                {this.state.onWatchlist === true && <button className="on-watchlist-button" onClick={() => this.toggleWatchlist()}>On Watchlist</button>}
+                {this.state.onWatchlist === false && <button className="add-watchlist-button" onClick={() => this.toggleWatchlist()}>+ Add to Watchlist</button>}
```

Comment:
These onClicks don't need to be anonymous functions:
`onClick={() => this.toggleWatchlist()}` => `onClick={this.toggleWatchlist}`